map.go

```diff
@@ -376,12 +384,9 @@ func (spec *MapSpec) createMap(inner *sys.FD, opts MapOptions) (_ *Map, err erro
 		spec.KeySize = 4
 		spec.ValueSize = 4
 
-		if spec.MaxEntries == 0 {
```

Comment:
This will leave a 0 in spec.MaxEntries if getting possible CPUs fails. I know it's annoying, but we shouldn't do this. (also the Go tenet of not using the other value(s) if err != nil, etc..)

Reply:
Can't remember what the proposed change looked like before, not sure how I missed the copy. I guess it would've been fine.

My point was: if a function returns a non-nil error, all other return values (other than the error itself, of course) should be ignored. IIRC this said something like `spec.MaxEntries, err = ...`, so I was just pointing out the obvious.